src/Binding/JSModules.cpp

```diff
@@ -403,7 +404,9 @@ JSModule::~JSModule()
     if (m_FilePath && m_Cached) {
         m_Modules->remove(this);
     }
-
+    if (m_dlmodule && m_ModuleType == kModuleType_Native) {
```

Comment:
No need to check if the module is native or not. Checking that `m_dlmodule` is not null is enough.